sql/plan/load_data.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+)
+
+type LoadData struct {
+	Local              bool
+	File               string
+	Destination        sql.Node
+	ColumnNames        []string
+	ResponsePacketSent bool
+	Fields             *sqlparser.Fields
+	Lines              *sqlparser.Lines
+	IgnoreNum          int64
+}
+
+const (
+	Tmpfiledir  = "/tmp/"
+	TmpfileName = ".LOADFILE"
+)
+
+var (
+	fieldsTerminatedByDelim = "\t"
+	fieldsEnclosedByDelim   = ""
+	fieldsOptionallyDelim   = false
+	fieldsEscapedByDelim    = "\\"
+	linesTerminatedByDelim  = "\n"
+	linesStartingByDelim    = ""
+)
+
+func (l *LoadData) Resolved() bool {
+	return l.Destination.Resolved()
+}
+
+func (l *LoadData) String() string {
+	pr := sql.NewTreePrinter()
+
+	_ = pr.WriteNode("LOAD DATA")
+	_ = pr.WriteChildren(l.Destination.String())
+	return pr.String()
+}
+
+func (l *LoadData) Schema() sql.Schema {
+	return l.Destination.Schema()
+}
+
+func (l *LoadData) Children() []sql.Node {
+	return []sql.Node{l.Destination}
+}
+
+func (l *LoadData) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
```

Comment:
You want to scan file lines one at a time as you load rows, not all at once. Otherwise we read the entire file before we begin making progress inserting rows, and we need the entire file in memory (which may be massive). So open the file and scanner in RowIter, then return a sql.RowIter implementation that reads lines off of it one at a time.